src/mcp/index.ts

```diff
@@ -499,6 +508,18 @@ export class FirebaseMcpServer {
       return;
     }
 
-    if (this._ready) void this.server.sendLoggingMessage({ level, data });
+    if (this._ready) {
+      // once ready, flush all pending messages before sending the next message
+      // this should only happen during startup
+      while (this._pendingMessages.length) {
+        const message = this._pendingMessages.shift();
+        if (!message) continue;
+        this.log(message.level, message.data);
```

Comment:
Why recurse instead of calling sendLoggingMessage here?

Reply:
the robots - it shouldn't :)